core/src/main/java/org/infinispan/marshall/exts/ThrowableExternalizer.java

```diff
@@ -0,0 +1,291 @@
+package org.infinispan.marshall.exts;
+
+
+import java.io.IOException;
+import java.io.ObjectInput;
+import java.io.ObjectOutput;
+import java.lang.reflect.InvocationTargetException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.TimeoutException;
+
+import org.infinispan.IllegalLifecycleStateException;
+import org.infinispan.InvalidCacheUsageException;
+import org.infinispan.commons.CacheConfigurationException;
+import org.infinispan.commons.CacheException;
+import org.infinispan.commons.CacheListenerException;
+import org.infinispan.commons.dataconversion.EncodingException;
+import org.infinispan.commons.marshall.AdvancedExternalizer;
+import org.infinispan.commons.marshall.MarshallUtil;
+import org.infinispan.commons.marshall.MarshallingException;
+import org.infinispan.commons.util.Util;
+import org.infinispan.interceptors.distribution.ConcurrentChangeException;
+import org.infinispan.interceptors.impl.ContainerFullException;
+import org.infinispan.manager.EmbeddedCacheManagerStartupException;
+import org.infinispan.marshall.core.Ids;
+import org.infinispan.notifications.IncorrectListenerException;
+import org.infinispan.partitionhandling.AvailabilityException;
+import org.infinispan.persistence.spi.PersistenceException;
+import org.infinispan.remoting.CacheUnreachableException;
+import org.infinispan.remoting.RemoteException;
+import org.infinispan.remoting.RpcException;
+import org.infinispan.remoting.transport.jgroups.SuspectException;
+import org.infinispan.statetransfer.AllOwnersLostException;
+import org.infinispan.statetransfer.OutdatedTopologyException;
+import org.infinispan.topology.CacheJoinException;
+import org.infinispan.transaction.WriteSkewException;
+import org.infinispan.transaction.xa.InvalidTransactionException;
+import org.infinispan.util.UserRaisedFunctionalException;
+import org.infinispan.util.concurrent.locks.DeadlockDetectedException;
+
+public class ThrowableExternalizer implements AdvancedExternalizer<Throwable> {
```

Comment:
I guess I never thought about it before, but this means we were using default Java serialization for Exceptions before? Technically this could have been done in a different PR though (although I wouldn't push that atm).

Either way great catch!

Reply:
> I guess I never thought about it before, but this means we were using default Java serialization for Exceptions before?

Yep

Like the `ImmutableListCopy.Externalizer` this isn't strictly needed for the JavaSerializationMarshaller to work, but is for other non-serializable marshallers.